From: olh@suse.de
Subject: no binary garbage in md dump

The kernel considers 0xc0-0xff as printable chars, this leads to
binary garbage. Just mask it to ASCII.

Index: linux-2.6.5/kdb/kdbmain.c
===================================================================
--- linux-2.6.5.orig/kdb/kdbmain.c
+++ linux-2.6.5/kdb/kdbmain.c
@@ -2091,7 +2091,7 @@ kdb_md_line(const char *fmtstr, kdb_mach
 			cp = wc.c;
 #endif
 			wc.word = word;
-#define printable_char(c) ({unsigned char __c = c; isprint(__c) ? __c : '.';})
+#define printable_char(c) ({unsigned char __c = toascii(c); isalnum(__c) ? __c : '.';})
 			switch (bytesperword) {
 			case 8:
 				*c++ = printable_char(*cp++);
